pkg/schedule/windowed_schedule.go

```diff
@@ -49,12 +105,24 @@ func (w *WindowedSchedule) Validate() error {
 	if w.Interval <= 0 {
 		return ErrInvalidInterval
 	}
+
+	// the schedule passed validation, set as active
+	w.state = Active
```

Comment:
**To reviewers**:
Setting schedule state as active is very important in a scenario when the task is restarted, so validation happens once again to check if the schedule is still valid at this point in time. This is the best place to set schedule's state to active if validation is passed. The only active schedule might be fired again.